src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -0,0 +1,562 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class PreferStreamReadAsyncMemoryOverloadsTest : PreferStreamAsyncMemoryOverloadsTestBase
+    {
+        #region C# - No diagnostic - Analyzer
+
+        [Fact]
+        public Task CS_Analyzer_NoDiagnostic_Read()
+        {
+            return AnalyzeCSAsync(@"
+using System.IO;
+class C
+{
+    public void M()
+    {
+        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
+        {
+            byte[] buffer = new byte[fs.Length];
+            fs.Read(buffer, 0, (int)fs.Length);
+        }
+    }
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_Analyzer_NoDiagnostic_ReadAsync_ByteMemory()
+        {
+            return AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+class C
+{
+    public async void M()
+    {
+        using (FileStream fs = new FileStream(""path.txt"", FileMode.Create))
+        {
+            byte[] buffer = new byte[fs.Length];
+            Memory<byte> memory = new Memory<byte>(buffer);
+            await fs.ReadAsync(memory, new CancellationToken()).ConfigureAwait(false);
+        }
+    }
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_Analyzer_NoDiagnostic_ReadAsync_AsMemory()
+        {
+            return AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+class C
+{
+    public async void M()
+    {
+        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
+        {
+            byte[] buffer = new byte[fs.Length];
+            await fs.ReadAsync(buffer.AsMemory(), new CancellationToken());
+        }
+    }
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_Analyzer_NoDiagnostic_NoAwait_SaveAsTask()
+        {
+            return AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    public void M()
+    {
+        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
+        {
+            byte[] buffer = new byte[fs.Length];
+            Task t = fs.ReadAsync(buffer, 0, (int)fs.Length);
+        }
+    }
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_Analyzer_NoDiagnostic_NoAwait_ReturnMethod()
+        {
+            return AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    public Task M(FileStream fs, byte[] buffer)
+    {
+        return fs.ReadAsync(buffer, 0, (int)fs.Length);
```

Comment:
Consider one that also ignores the result here. just

```C#
public void M(Stream s, byte[] buf)
{
    s.ReadAsync(buf, 0, 1);
}
```

Because that will change the parent expression hierarchy (parent might be null here? or it might be "expression statement")

Reply:
Added the unit test with no diagnostic expected. The test passed.

Reply:
Added. It passed.